### ISSUE TITLE
Edit Message with multiple lines broken

### ISSUE THREAD

**Issue by adamkerz** [sublimehq/sublime_merge#1358]
**Version info**

 - OS: Windows 10
 - Build: 2063 (works on 2059)

**Description**

Editing a commit message and trying to save a message with multiple lines results in an error.

**Steps to reproduce**

1. Open repository <any> (I've tried it on two unrelated repos)
2. Run command Edit Commit Message
3. Click on Save

**Expected behavior**

The message is saved.

**Screenshots**

![multiline_message](https://user-images.githubusercontent.com/7362087/145942344-c7d49b7b-7658-4e34-a75c-78de70029592.png)

**Comment by srbs** [comment]
@adamkerz, can you show the command that was run? (click the `>!` to the left of the `failed:`)

**Comment by dpjohnst** [comment]
G'day @adamkerz,

Thanks for reaching out - it sounds like the commit editing operation is failing to create the new commit object.
This can be for a number of reasons, so I'll be investigating further.

When you have a spare moment, would you mind sharing your Sublime Merge debug information.
This can be found in the application hamburger menu under `Help > Debug Information`.

Before sharing the information here, please take a moment to remove any information you feel is sensitive such as file names, author information, remote URLs etc.

Additionally, could you run the following command in the repository via the command line and share the output:
`git commit-tree -m "Test Message" -p e7c4602d4a 1acaaf2905`

Kind regards,
\- Dylan from Sublime HQ

**Comment by adamkerz** [comment]
It just says `Edit Commit`, no command...

That was a test commit that I blew away after. But the output of that command is:
```
> git commit-tree -m "Test Message" -p e7c4602d4a 1acaaf2905
c07d1766aa91882b15c63baeaaa3f3a03605e758
```

I re-created a similar test message and re-ran the command:

![new](https://user-images.githubusercontent.com/7362087/146105904-f718aebf-22a2-448d-9ff5-68ef2998ecae.png)

```
> git commit-tree -m "Test Message" -p e7c4602d4a ddfad739ac
fatal: ddfad739acfe2c0c7ebab0936d1e845df26d82a3 is not a valid 'tree' object
```

**Comment by adamkerz** [comment]
```
=== App Version Information ===
Build: 2063

=== Git Version Information ===
Using Git: D:\pe\apps-core\bin\git.bat (custom)
git version 2.28.0.windows.1
PATH: C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\TortoiseGit\bin;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\Microsoft\Azure Functions Core Tools\;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;D:\pe\apps-core\bin;%USERPROFILE%\AppData\Local\Microsoft\WindowsApps;;D:\Fiddler

=== Browse Page Information ===
HEAD: ddfad739acfe2c0c7ebab0936d1e845df26d82a3
Is in merge: 0
Is in cherry_pick: 0
Is in rebase: 0
Is in revert: 0

=== Git Status Information ===
?? database/database_backups/2021-12-13.sql
?? src/routes/membership/membership.code_to_add_six_months_or_year.js

=== Our Status Information ===
?? database/database_backups/2021-12-13.sql
?? src/routes/membership/membership.code_to_add_six_months_or_year.js

=== Git Config Information ===
branch.master.merge=refs/heads/master
branch.master.remote=origin
color.branch=auto
color.diff=auto
color.interactive=true
color.status=auto
core.autocrlf=false
core.bare=false
core.filemode=false
core.fscache=true
core.ignorecase=true
core.logallrefupdates=true
core.repositoryformatversion=0
core.symlinks=false
credential.helper=wincred
diff.astextplain.textconv=astextplain
filter.lfs.clean=git-lfs clean -- %f
filter.lfs.process=git-lfs filter-process
filter.lfs.required=true
filter.lfs.smudge=git-lfs smudge -- %f
help.format=html
http.proxy=http://127.0.0.1:38080
rebase.autosquash=true
remote.github.fetch=+refs/heads/*:refs/remotes/github/*
remote.github.tagopt=--no-tags
remote.github.url=https://github.com/xxxx/qcrc-website
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
remote.origin.puttykeyfile=D:\xxxx@kerz.id.au.ppk
remote.origin.url=ssh://git@bitbucket.org/xxxx/qcrc-website.git
user.email=qutCliffhangers@kerz.id.au
user.name=Adam Kerz

=== Our Config Information ===
Git Config Path Information
Using config path: /C/ProgramData/Git/config
Using config path: /D/pe/apps-core/mingw64/etc/gitconfig
Using config path: /D/pe/home/.config/git/config
Using config path: /D/pe/home/.gitconfig
Using config path: /D/development/xxxx/qcrc-website/.git/config

Our config output is different to the Git config output
branch.master.merge=refs/heads/master
branch.master.remote=origin
core.autocrlf=false
core.bare=false
core.filemode=false
core.ignorecase=true
core.logallrefupdates=true
core.repositoryformatversion=0
core.symlinks=false
credential.helper=wincred
http.proxy=http://127.0.0.1:38080
remote.github.fetch=+refs/heads/*:refs/remotes/github/*
remote.github.tagopt=--no-tags
remote.github.url=https://github.com/xxxx/qcrc-website
remote.origin.fetch=+refs/heads/*:refs/remotes/origin/*
remote.origin.puttykeyfile=D:\xxxx@kerz.id.au.ppk
remote.origin.url=ssh://git@bitbucket.org/xxxx/qcrc-website.git
user.email=qutCliffhangers@kerz.id.au
user.name=Adam Kerz

=== Git Attributes Information ===
git check_attr --all output

=== Our Modified Files Newline Normalisation and EOL Information ===

=== Our Modified Files Flag Information ===
Ignoring symlinks: 1
``

**Comment by adamkerz** [comment]
Looks like this might have been caused by the git.bat file I was using to set some env vars before running git. Fixed as part of debugging #1359, bug brought back by pointing back at the .bat file. 100% the cause, not the older version of git (still using git version 2.28.0.windows.1) that I also noticed in the debug output.
